File: 프로그래머스/unrated/181885. 할 일 목록/할 일 목록.java
import java.util.*;
class Solution {
    public String[] solution(String[] todo_list, boolean[] finished) {
        List<String> list = new ArrayList<>();
        for(int i = 0 ; i<finished.length ;i++){
            if(!finished[i]) list.add(todo_list[i]);
        }
        int idx = 0;
        String[] answer = new String[list.size()];
        for(String s : list){
            answer[idx++] = s; 
        }
        return answer;
    }
}

File: 프로그래머스/unrated/181885. 할 일 목록/할 일 목록.js
function solution(todo_list, finished) {
    var answer = [];
    let idx = 0;
    for(let i = 0 ; i < todo_list.length ; i++){
        if(!finished[i]){
            answer[idx++] = todo_list[i];
        }
    }
    return answer;
}